client/capiclient/clusters/clusters_client.go

```diff
@@ -0,0 +1,98 @@
+package clusters
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/gsclientgen/models"
+	"github.com/giantswarm/microerror"
+	"github.com/go-openapi/strfmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	clusterMaxAge = 24 * time.Hour
+)
+
+type Client struct {
+	formats strfmt.Registry
+}
+
+func New(formats strfmt.Registry) *Client {
+	client := &Client{
+		formats: formats,
+	}
+
+	return client
+}
+
+// Using response model from gs api until it's deprecated, for compatibility
+func (c *Client) GetClusters(clientset *versioned.Clientset) ([]*models.V4ClusterListItem, error) {
+	clustersV4, err := clientset.CoreV1alpha1().AWSClusterConfigs(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	clustersV5, err := clientset.InfrastructureV1alpha2().AWSClusters(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var (
+		deleteDate time.Time
+
+		tClusters              = len(clustersV4.Items) + len(clustersV5.Items)
+		payload                = make([]*models.V4ClusterListItem, 0, tClusters)
+	)
+
+	// V4 Clusters
+	for _, cluster := range clustersV4.Items {
+		formattedCluster := &models.V4ClusterListItem{
+			ID:             cluster.Spec.Guest.ID,
+			CreateDate:     cluster.GetCreationTimestamp().UTC().Format(time.RFC3339),
+			Name:           cluster.Spec.Guest.Name,
+			Owner:          cluster.Spec.Guest.Owner,
+			ReleaseVersion: cluster.Spec.Guest.ReleaseVersion,
+		}
+		formattedCluster.Path = fmt.Sprintf("/v4/clusters/%s/", formattedCluster.ID)
+		if cluster.DeletionTimestamp != nil {
+			deleteDate = cluster.GetDeletionTimestamp().Time.UTC()
+			if time.Now().Sub(deleteDate) < clusterMaxAge {
```

Comment:
Are you replicating some logic that was in the cluster listing before? I think we don't need to filter out old deleted clusters.

We should show them, since it means there is a bug, and we'd actually want users to reach out to us (or for us to be able to notice them) when listing clusters using our interfaces.

Reply:
The GiantSwarm API cluster lister response only includes clusters that are ~1 day old. I haven't been able to find the place where that filter is applied. The reason for adding the filter is that the output of the new kubernetes client output included more clusters than the GS API way

Reply:
Hmm.. in your screenshot, there appear to be some repeated cluster ID's in there, not sure if that is influencing your comparison between old gsctl output and this.

I can't recall that we are filtering deleted clusters out by some max age.

Reply:
Don't know where the duplication came from, they only get displayed once now (this is with filtering off)
![image](https://user-images.githubusercontent.com/13508038/78553997-90a59c00-780a-11ea-834d-6eb73af3f415.png)

GSCTL output
![image](https://user-images.githubusercontent.com/13508038/78554146-e9753480-780a-11ea-8d78-adeb78d142a6.png)

The main difference is from some clusters that have been marked as deleted a long time ago, that don't show up in `gsctl`.

Also, that only happens for V4 clusters.